src/dali/plugin/input/csv/nexo.py

```diff
@@ -0,0 +1,150 @@
+
+import logging
+from csv import reader
+from typing import Dict, List, Optional
+
+from rp2.logger import create_logger
+from rp2.rp2_decimal import RP2Decimal
+
+from dali.abstract_input_plugin import AbstractInputPlugin
+from dali.abstract_transaction import AbstractTransaction
+from dali.configuration import Keyword
+from dali.in_transaction import InTransaction
+from dali.intra_transaction import IntraTransaction
+from dali.out_transaction import OutTransaction
+
+# _ACH_DEPOSIT: str = "Ach Deposit"
+# _ACH_WITHDRAWAL: str = "Ach Withdrawal"
+# _BUY_CURRENCY: str = "Buy Currency"
+# _BUY_QUANTITY: str = "Buy Quantity"
+# _CRYPTO_TRANSFER: str = "Crypto Transfer"
+# _DATE: str = "Date"
+# _INTEREST_PAYMENT: str = "Interest Payment"
+# _REFERRAL_BONUS: str = "Referral Bonus"
+# _SOLD_CURRENCY: str = "Sold Currency"
+# _SOLD_QUANTITY: str = "Sold Quantity"
+# _TRADE: str = "Trade"
+# _TRADE_ID: str = "Trade ID"
+# _TYPE: str = "Type"
+# _WITHDRAWAL: str = "Withdrawal"
+# _WITHDRAWAL_FEE: str = "Withdrawal Fee"
+
+
+# types of transactions
+_INTEREST = "Interest"
+_LOCKING_TERM_DEPOSIT = "LockingTermDeposit"
+_UNLOCKING_TERM_DEPOSIT = "UnlockingTermDeposit"
+_FIXED_TERM_INTEREST = "FixedTermInterest"
+_DEPOSIT = "Deposit"
+
+class InputPlugin(AbstractInputPlugin):
+
+    __NEXO: str = "Nexo"
+
+    __TRANSACTION_ID_INDEX = 0
+    __TRANSACTION_TYPE_INDEX = 1
+    __CURRENCY_INDEX: int = 2
+    __AMOUNT_INDEX: int = 3
+
+    __TIMESTAMP_INDEX: int = 7
+
+    __DELIMITER = ","
+
+    def __init__(
+        self,
+        account_holder: str,
+        transaction_csv_file: str,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder=account_holder, native_fiat=native_fiat)
+        self.__transaction_csv_file: str = transaction_csv_file
+        self.__logger: logging.Logger = create_logger(f"{self.__NEXO}/{self.account_holder}")
+
+    def load(self) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        last_withdrawal_fee: Optional[RP2Decimal] = None
+        with open(self.__transaction_csv_file, encoding="utf-8") as transaction_csv_file:
+            # read CSV with header and skip first row
+            lines = reader(transaction_csv_file)
+
+            # Skip header line
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                transaction_id: str = line[self.__TRANSACTION_ID_INDEX].strip()
+                transaction_type: str = line[self.__TRANSACTION_TYPE_INDEX].strip()
+                currency: str = line[self.__CURRENCY_INDEX].strip()
+                amount = line[self.__AMOUNT_INDEX].strip()
+                timestamp_with_timezone = f"{line[self.__TIMESTAMP_INDEX].strip()} -00:00"
```

Comment:
This line suggests the timestamp in the CSV has no timezone information, but the timezone is always assumed to be UTC. Is this correct?

Reply:
I think that's right. I'll confirm and document in-code.